akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala

```diff
@@ -550,41 +206,39 @@ private[akka] class StreamSupervisor(settings: ActorFlowMaterializerSettings) ex
  * INTERNAL API
  */
 private[akka] object ActorProcessorFactory {
+  import akka.stream.impl.Stages._
 
-  import Ast._
-  def props(materializer: ActorFlowMaterializer, op: AstNode): Props = {
-    val settings = materializer.settings // USE THIS TO AVOID CLOSING OVER THE MATERIALIZER BELOW
+  def props(materializer: ActorFlowMaterializerImpl, op: StageModule, parentAttributes: OperationAttributes): (Props, Any) = {
+    val att = parentAttributes and op.attributes
```

Comment:
this line and the one below is an actual fix, and the problem was exposed by the supervision additions.